### ISSUE TITLE
Usability issues for screen reader users (settings page)

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#8948]
This issue describes some inconveniences I have found by testing the Scratch Addons settings page using a screen reader. This involved navigating Scratch Addons with only a keyboard and Windows Narrator with the screen blacked out, simulating what a blind user would experience.

<img align="right" width="129" height="119" alt="Paintbrush and arrow icon on an addon" src="https://github.com/user-attachments/assets/dea10d62-706d-4c54-add7-25a699087beb" />

<h4>

- [ ] Some undescribed graphics

</h4>

Occasionally there are icons that have no image description, each one announced as just "graphic". These icons include:
- [ ] Addon category icons, which should indicate the type of an addon (whether an addon is for the editor or for the website, a feature or a theme, etc.) but currently convey no information to screen reader users.
- [ ] Alert icons, which indicate whether an addon alert is a generic, notice, or warning message, but only if you can see them.
- [ ] Help icons, which open descriptions of settings, but only if hovered over with the mouse or focused with the <kbd>Tab</kbd> key specifically (which is weird since icons shouldn't have an action). There is a container which, when focused, reveals the setting description, allowing it to be read, but for some reason it isn't highlighted when using a screen reader.
- [ ] Group icons in More Settings, which have no meaning and are just clutter.

Some of these are fixed by #8718.

<h4>

- [ ] Some unlabeled controls

</h4>

Some buttons such as "expand/collapse" toggles as well as checkboxes are not tied to their labels, so you don't know what they do unless you use context clues to identify the purpose of the button. It's acceptable once you're used to the layout but is still something that could be improved.

This also means that every one of these controls has at least two separate elements, the label and the input, which means more content to process and navigate through.

<h4>

- [ ] No landmarks

</h4>

There are no defined landmark elements or page regions to jump between, so if you're using the settings page and are deep in the addons container, you'll have to wade through it to get back to the navigation bar, for instance, unless your technology gives you a shortcut to an element on it like a heading or a link.

Landmarks are added to the main settings page in #9008.

<h4>

- [x] Table items can't be reordered

</h4>

Moving them requires a mouse.

Fixed by #8934.

---

There are also some things that, if done, would make navigation and use easier in my opinion:

<h4>

- [ ] Content groups

</h4>

There currently aren't any notable defined lists, groups, or invisible headings that a screen reader can announce, which means a user won't know where they are in the addons container, setting groups, or tables. Some of these are understandable; lists can have advantages if labeled well, but I understand if the development cost would outweigh the potential usability bonuses.

<h4>

- [ ] Content trees

</h4>

Think of the addons container like a folder structure, where first you open the group you want (recommended/featured/other/etc.), then select the addon, and finally read about it or change its settings. You can easily go up and down levels. Now look at how the settings page is right now and find that all of these are combined into one page where you can see everything at once.

Collapsing every item by default is one step towards a more hierarchal navigation style, which can help a user focus but takes more actions to navigate.

---

I'd love for some contributors to join me in an accessibility improvement journey, even just by testing and reviewing.

I know a bit about using [ARIA attributes](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA), so I can help with accessible UX.

**Comment by DNin01** [comment]
@WorldLanguages Do you know anyone who expresses an interest in web accessibility and might be open to test and review such updates?

**Comment by mxmou** [comment]
I can review your PRs once the new Blockly work is finished.

**Comment by WorldLanguages** [comment]
> [@WorldLanguages](https://github.com/WorldLanguages) Do you know anyone who expresses an interest in web accessibility and might be open to test and review such updates?

No one comes to mind. But that shouldn't be a problem. Even if you don't find a reviewer, accessibility enhancements should be merged anyway.
